From: Sascha Hauer <s.hauer@pengutronix.de>
Date: Mon, 25 Jul 2016 10:14:15 +0200
Subject: [PATCH] lib: Add Kconfig symbol for FNMATCH

fnmatch is useful on its own, so make a separate Kconfig symbol
and select it from GLOB.

Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
---
 common/Kconfig | 1 +
 lib/Kconfig    | 3 +++
 lib/Makefile   | 2 +-
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/common/Kconfig b/common/Kconfig
index 832634047fe4..8230e4bdef8a 100644
--- a/common/Kconfig
+++ b/common/Kconfig
@@ -424,6 +424,7 @@ config MAXARGS
 
 config GLOB
 	bool
+	select FNMATCH
 	prompt "globbing support"
 	help
 	  If you want to use wildcards like * or ? say y here.
diff --git a/lib/Kconfig b/lib/Kconfig
index d5f99ae2e599..f9f25bdef8f5 100644
--- a/lib/Kconfig
+++ b/lib/Kconfig
@@ -39,6 +39,9 @@ config BCH
 config BITREV
 	bool
 
+config FNMATCH
+	bool
+
 config QSORT
 	bool
 
diff --git a/lib/Makefile b/lib/Makefile
index 92404fd331f8..1be1742499b8 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -27,7 +27,7 @@ obj-$(CONFIG_ZLIB)	+= decompress_inflate.o zlib_inflate/
 obj-$(CONFIG_XZ_DECOMPRESS) += decompress_unxz.o xz/
 obj-$(CONFIG_CMDLINE_EDITING)	+= readline.o
 obj-$(CONFIG_SIMPLE_READLINE)	+= readline_simple.o
-obj-$(CONFIG_GLOB)		+= fnmatch.o
+obj-$(CONFIG_FNMATCH)		+= fnmatch.o
 obj-$(CONFIG_GENERIC_FIND_NEXT_BIT) += find_next_bit.o
 obj-y			+= glob.o
 obj-y			+= notifier.o
